controllers/sparkcluster_controller.go

```diff
@@ -447,13 +468,51 @@ func (r *SparkClusterReconciler) reconcileStatefulSets(ctx context.Context, sc *
 		log.V(1).Info("updating status", "path", ".status.workerSelector", "value", sc.Status.WorkerSelector)
 	}
 
+	var status v1.JobConditionType
+	masterPod, masterPodFound := r.getMasterPod(ctx, sc)
+	if masterPodFound {
+		if dcv1alpha1.IsPodReady(masterPod) {
```

Comment:
Does the Pod have the same health check defined on it that Nucleus uses? I wouldn't want to change the when the Pod is considered ready.

Reply:
Nop, Nucleus appears to use the same logic:

`KubernetesApiHelper.getPodConditions(pod).exists(condition => condition.getType == "Ready" && condition.getStatus == "True")`